modAionImpl/src/org/aion/zero/impl/AionHub.java

```diff
@@ -122,21 +125,35 @@ public static AionHub inst() {
     private static final int INIT_ERROR_EXIT_CODE = -1;
 
     public AionHub() {
+        initializeHub(CfgAion.inst(), AionBlockchainImpl.inst(), AionRepositoryImpl.inst(), false);
+    }
+
+    private void initializeHub(
+            CfgAion _cfgAion,
+            AionBlockchainImpl _blockchain,
+            AionRepositoryImpl _repository,
+            boolean forTest) {
 
-        this.cfg = CfgAion.inst();
+        this.cfg = _cfgAion;
 
         // load event manager before init blockchain instance
-        loadEventMgr();
+        loadEventMgr(forTest);
 
-        AionBlockchainImpl blockchain = AionBlockchainImpl.inst();
+        AionBlockchainImpl blockchain = _blockchain;
         blockchain.setEventManager(this.eventMgr);
         this.blockchain = blockchain;
 
-        this.repository = AionRepositoryImpl.inst();
+        this.repository = _repository;
 
-        this.mempool = AionPendingStateImpl.inst();
+        this.mempool =
+                forTest
+                        ? AionPendingStateImpl.createForTesting(_cfgAion, _blockchain, _repository)
+                        : AionPendingStateImpl.inst();
```

Comment:
Why not make mempool also injectable, by adding a `IPendingStateInternal<AionBlock, AionTransaction>` to the package-private constructor?

Then `createForTesting` can pass in `AionPendingStateImpl.createForTesting(_cfgAion, _blockchain, _repository)`, and the "normal" code path can pass in `AionPendingStateImpl.inst()`